### ISSUE TITLE
linter: Nested configs are not working with VSCode extension

### ISSUE THREAD

**Issue by uzmoi** [oxc-project/oxc#9899]
### What version of Oxlint are you using?

0.16.0

### What command did you run?

(VSCode extension 0.16.0)

### What does your `.oxlintrc.json` config file look like?

```jsonc
// .oxlintrc.json
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "rules": { "no-eval": "warn" }
}
```
```jsonc
// foo/.oxlintrc.json
{
  "$schema": "../node_modules/oxlint/configuration_schema.json",
  "rules": { "no-eval": "off" }
}
```

### What happened?

Seems like oxlint is using .oxlintrc.json in the workspace root directory instead of foo/.oxlintrc.json to lint foo/index.ts.

![Image](https://github.com/user-attachments/assets/a88314a7-3c86-43cd-b47e-32ba303fb64c)

I think this is a VSCode extension problem because when I run `npx oxlint` in terminal, no warnings are found (`Found 0 warnings and 0 errors.`).

**Comment by camchenry** [comment]
This should be fixed in the next release of oxlint, as we hadn't merged LSP+editor support for this in 0.16.0. https://github.com/oxc-project/oxc/pull/9743

**Comment by Boshen** [comment]
Released in https://github.com/oxc-project/oxc/releases/tag/oxlint_v0.16.1
